runtime-modules/referendum/src/benchmarking.rs

```diff
@@ -0,0 +1,739 @@
+                current_cycle_id: 0,
+            }
+        );
+
+        move_to_block::<T, I>(
+            target_block_number,
+            target_stage
+        );
+    }: { Referendum::<T, I>::on_finalize(System::<T>::block_number()); }
+    verify {
+        assert_eq!(
+            Referendum::<T, I>::stage(),
+            ReferendumStage::Inactive,
+            "Reveal perdiod hasn't ended",
+        );
+
+        assert_last_event::<T, I>(RawEvent::ReferendumFinished(intermediate_winners).into());
+    }
+
+    on_finalize_voting {
+        let winning_target_count = 0;
+        start_voting_cycle::<T, I>(winning_target_count);
+
+        let started_voting_block_number = System::<T>::block_number() + One::one();
+        let target_block_number =
+            T::VoteStageDuration::get() + started_voting_block_number - One::one();
+
+        let target_stage = ReferendumStage::Voting(ReferendumStageVoting {
+                started: System::<T>::block_number() + One::one(),
+                winning_target_count: (winning_target_count + 1).into(),
+                current_cycle_id: 0,
+        });
+
+        move_to_block::<T, I>(target_block_number, target_stage);
+    }: { Referendum::<T, I>::on_finalize(System::<T>::block_number()); }
+    verify {
+        let current_stage = ReferendumStage::Revealing(ReferendumStageRevealing {
+            started: target_block_number + One::one(),
+            winning_target_count: 1,
+            intermediate_winners: vec![],
+            current_cycle_id: 0,
+        });
+
+        assert_eq!(
+            Referendum::<T, I>::stage(),
+            current_stage,
+            "Voting period not ended"
+        );
+
+        assert_last_event::<T, I>(RawEvent::RevealingStageStarted().into());
+    }
+
+    vote {
+        start_voting_cycle::<T, I>(0);
+
+        let account_id = funded_account::<T, I>("caller", 0);
+
+        let salt = vec![0u8];
+        let cycle_id = 0;
+        let vote_option = 0;
+        let commitment =
+            Referendum::<T, I>::calculate_commitment(
+                &account_id,
+                &salt,
+                &cycle_id,
+                &vote_option.into()
+            );
+
+        let stake = T::MinimumStake::get() + One::one();
+    }: _ (RawOrigin::Signed(account_id.clone()), commitment, stake)
+    verify {
+        assert!(Votes::<T, I>::contains_key(account_id.clone()), "Vote wasn't added");
+
+        assert_eq!(
+            Referendum::<T, I>::votes(account_id.clone()),
+            CastVote {
+                commitment,
+                stake,
+                cycle_id,
+                vote_for: None,
+            },
+            "Vote is not correctly Stored",
+        );
+
+        assert_last_event::<T, I>(RawEvent::VoteCast(account_id.clone(), commitment, stake).into());
+    }
+
+    reveal_vote_space_for_new_winner {
+        let i in 0 .. MAX_WINNERS;
+
+        let salt = vec![0u8];
+        let vote_option = 2 * (i + 1);
```

Comment:
What does this formula mean? Does it mean "greater than voters number"?

Reply:
It means "greater than voters + candidates" because I use the 1 voter for each candidate `add_and_reveal_multiple_votes_and_add_extra_unrevealed_vote`